lib/knex/db.ts

```diff
@@ -756,15 +756,15 @@ module.exports = function (PersistObjectTemplate) {
                     else
                         table[type](columns, name);
 
-                }).bind(this));
+                }));
             }
 
 
             return knex.transaction(function (trx) {
                 return trx.schema.table(tableName, function (table) {
-                    _.map(Object.getOwnPropertyNames(dbChanges), function (key) {
-                        return syncIndexesForHierarchy.call(this, key, dbChanges, table);
-                    }.bind(this));
+                    syncIndexesForHierarchy('delete', dbChanges, table);
```

Comment:
is there any test in persister that verifies that this works (changing an index type)?